worker/src/RTC/Transport.cpp

```diff
@@ -947,13 +985,13 @@ namespace RTC
 
 			case Channel::Request::MethodId::TRANSPORT_PRODUCE_DATA:
 			{
-				if (!this->sctpAssociation)
+				if (!this->sctpAssociation && !this->direct)
 					MS_THROW_ERROR("SCTP not enabled");
 
 				std::string dataProducerId;
 
 				// This may throw.
-				SetNewDataProducerIdFromRequest(request, dataProducerId);
+				SetNewDataProducerIdFromInternal(request->internal, dataProducerId);
 
 				// This may throw.
 				auto* dataProducer = new RTC::DataProducer(dataProducerId, this, request->data);
```

Comment:
Verify that producer type corresponds with transport type.

Reply:
Done.